### PR TITLE
add stage-x presets

### PR DESCRIPTION
**Pull request by hzoo**
Ref https://github.com/babel/babel/issues/2626

Not sure we want to do it this way but figured I could just make the initial pr?

### CODE REVIEW

File: packages/babel-preset-stage-1/index.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  presets: [
+    require("babel-preset-stage-2")
+  ],
+  plugins: [
+    require("babel-plugin-syntax-class-constructor-call"),
+    require("babel-plugin-syntax-class-properties"),
+    require("babel-plugin-syntax-decorators"),
+    require("babel-plugin-syntax-export-extensions"),
+    // should be babel-plugin-transform-class-constructor-call
+    require("babel-plugin-transform-class-construtor-call"),
```
Comment (sebmck) [inline packages/babel-preset-stage-1/index.js]: This is fixed in `master` so the correct spelling can be used.

File: packages/babel-preset-stage-0/index.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  presets: [
+    require("babel-preset-stage-1")
+  ],
+  plugins: [
+    require("babel-plugin-syntax-do-expressions"),
+    require("babel-plugin-syntax-function-bind"),
+    require("babel-plugin-transform-do-expressions"),
```
Comment (sebmck) [inline packages/babel-preset-stage-0/index.js]: When you enable transforms it also enables the syntax so we don't need to double up.

Comment (hzoo) [inline packages/babel-preset-stage-0/index.js]: :dizzy_face: good to know - cool!

**Comment by hzoo** [comment]
Ok removed all syntax plugins except trailing commas (since that's all that's needed)

**Comment by hzoo** [comment]
should I add these in `babel-cli/test/index`

```js
var presetLocs = [
  __dirname + "/../../babel-preset-es2015",
  __dirname + "/../../babel-preset-react"
].join(",");
```

File: packages/babel-preset-stage-0/index.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  presets: [
+    require("babel-preset-stage-1")
```
Comment (sebmck) [inline packages/babel-preset-stage-0/index.js]: Stage 0 shouldn't include stage 1. Stage 0 is the earliest while stage 4 is the latest. You've got these the wrong way around.

Comment (thejameskyle) [inline packages/babel-preset-stage-0/index.js]: Huh? That seems correct. Stage 0 needs everything that's newer than it.

Comment (sebmck) [inline packages/babel-preset-stage-0/index.js]: Wait yeah. I need sleep lol.

**Comment by hzoo** [comment]
Should the presets be versioned starting at 1.0.0?

File: packages/babel-preset-stage-0/index.js (hunk 3 above)
Comment (hzoo) [inline packages/babel-preset-stage-0/index.js]: Insert @gaearon tweet about getting sleep :smile:

**Comment by sebmck** [comment]
They'll automatically be forced to 6.0.0 so probably not necessary. Thanks @hzoo! Can you also submit a docs PR?

**Comment by hzoo** [comment]
yeah got it

**Comment by hzoo** [comment]
Oh and someone will need to publish these later

**Pull request merged**